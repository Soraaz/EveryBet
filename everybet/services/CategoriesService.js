"use strict";

const HTTPClient = require("./HTTPClient");
const Common = require('./Common');

class CategoriesService {

    // POST

    static createNewCategory(name, icon) {
        let body = {
            name: name,
            icon: icon
        };
        return HTTPClient.sendPost("categories", body);
    }

    // PUT

    static updateCategoryById(id, columns, value) {
        if (arguments.length === 3) {
            let columnsMap = new Map();
            columnsMap.set(columns, value);
            columns = columnsMap;
        }
        return HTTPClient.sendPut("Categories/" + id, Common.parseColumns(columns));
    }

    static updateCategoriesByFilters(filters, columns) {
        return HTTPClient.sendPut(Common.parseFilters("Categories", filters), Common.parseColumns(columns));
    }

    static updateAllCategories(columns, value) {
        if (arguments.length === 2) {
            let columnsMap = new Map();
            columnsMap.set(columns, value);
            columns = columnsMap;
        }
        return this.updateCategoriesByFilters([], columns);
    }
    
    static updateCategoriesByName(name, columns, value) {
        let filters = new Map();
        filters.set("name", name);
        if (arguments.length === 3) {
            let columnsMap = new Map();
            columnsMap.set(columns, value);
            columns = columnsMap;
        }
        return this.updateCategoriesByFilters(filters, columns);
    }

    // DELETE

    static deleteCategorieById(id) {
        return HTTPClient.sendDelete("categories/" + id);
    }

    static deleteCategorieByFilters(filters) {
        return HTTPClient.sendDelete(Common.parseFilters("categories", filters));
    }

    // GET

    static getCategoryById(id) {
        return HTTPClient.sendGet("categories/" + id);
    }
    
    static getCategoriesByFilters(filters) {
        return HTTPClient.sendGet(Common.parseFilters("categories", filters));
    }

    static getAllCategories() {
        return this.getCategoriesByFilters([]);
    }

    static getCategoriesByName(name) {
        let filters = new Map();
        filters.set("name", name);
        return this.getCategoriesByFilters(filters);
    }
}

module.exports = CategoriesService;