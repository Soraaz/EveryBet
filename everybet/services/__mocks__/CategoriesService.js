"use strict";

class CategoriesService {
    static getAllCategories = jest.fn(() => {
        return new Promise(async (resolve) => {
            let res = [{
                "icon": "art.png",
                "id": 1,
                "name": "Art",
            }, {
                "icon": "finance.png",
                "id": 2,
                "name": "Finance",
            }, {
                "icon": "fun.png",
                "id": 3,
                "name": "Fun",
            }, {
                "icon": "literature.png",
                "id": 4,
                "name": "Littérature",
            }, {
                "icon": "news.png",
                "id": 5,
                "name": "News",
            }, {
                "icon": "numeric.png",
                "id": 6,
                "name": "Numérique",
            }, {
                "icon": "people.png",
                "id": 7,
                "name": "People",
            }, {
                "icon": "politics.png",
                "id": 8,
                "name": "Politique",
            }, {
                "icon": "science.png",
                "id": 9,
                "name": "Sciences",
            }, {
                "icon": "space.png",
                "id": 10,
                "name": "Espace",
            }, {
                "icon": "sport.png",
                "id": 11,
                "name": "Sport",
            }, {
                "icon": "video_games.png",
                "id": 12,
                "name": "Jeux Vidéos",
            }];
            resolve(res);
        });
    });
}

module.exports = CategoriesService;
