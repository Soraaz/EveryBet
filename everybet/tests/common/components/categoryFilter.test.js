import React from "react";
import {Platform} from "react-native";
import {shallow} from "enzyme";
import CategoriesService from "../../../services/CategoriesService";
import CategoriesServiceMock from "../../../services/__mocks__/CategoriesService";

import CategoryFilter from "../../../src/common/components/categoryFilter";
import renderer from "react-test-renderer";

describe("CategoryFilter", () => {
	beforeEach(() => {
		CategoriesService.getAllCategories = CategoriesServiceMock.getAllCategories;
	});

	describe("rendering", () => {
		/*		describe("android", () => {
					beforeAll(() => {
						Platform.OS = "android";
					});

					it("Should render correctly", () => {
						const tree = renderer
							.create(<CategoryFilter filterCallback={() => {
							}}/>)
							.toJSON();
						expect(tree).to.matchSnapshot();
					});
				});
		*/
		describe("ios", () => {
			beforeAll(() => {
				Platform.OS = "ios";
			});

			it("Should render correctly", () => {
				const tree = renderer
					.create(<CategoryFilter filterCallback={() => {}}/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});
		});
	});

	describe("functionalities", () => {
		it("Should set state on mount", (done) => {
			const wrapper = shallow(<CategoryFilter filterCallback={() => {}}/>);
			process.nextTick(() => {
				wrapper.update();
				expect(wrapper.state("categoryList").length).to.equal(12);
				done();
			});
		});

		it("Should set state on change category", (done) => {
			const wrapper = shallow(<CategoryFilter filterCallback={() => {}}/>);
			wrapper.instance()._changeCategory("sport");
			process.nextTick(() => {
				wrapper.update();
				expect(wrapper.state("category")).to.equal("sport");
				done();
			});
		});
	});
});
