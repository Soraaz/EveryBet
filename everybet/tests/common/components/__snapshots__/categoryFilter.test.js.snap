// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter rendering ios Should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "marginHorizontal": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
  >
    Cat√©gorie
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "height": 25,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingLeft": 10,
        }
      }
    >
      <Text>
        Toutes
      </Text>
    </View>
  </View>
</View>
`;
